vector/src/main/java/im/vector/riotredesign/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -35,21 +37,45 @@ import java.text.SimpleDateFormat
 import java.util.*
 
 
+val dateFormat = SimpleDateFormat("EEE, d MMM yyyy HH:mm", Locale.getDefault())
+
 data class MessageActionState(
         val roomId: String,
         val eventId: String,
         val informationData: MessageInformationData,
-        val userId: String = "",
-        val senderName: String = "",
-        val messageBody: CharSequence? = null,
-        val ts: String? = null,
-        val showPreview: Boolean = false,
-        val canReact: Boolean = false,
-        val senderAvatarPath: String? = null)
-    : MvRxState {
+        val timelineEvent: TimelineEvent?
+) : MvRxState {
+
+    fun senderName(): String = informationData.memberName?.toString() ?: ""
+
+    fun time(): String? = dateFormat.format(Date(timelineEvent?.root?.originServerTs ?: 0))
```

Comment:
if timelineEvent?.root?.originServerTs is null, we will display an Epoch time, which is not what we want. Maybe return "" in this case

Reply:
done